firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java

```diff
@@ -117,6 +117,13 @@ public boolean isCollectionGroupQuery() {
     return collectionGroup != null;
   }
 
+  /**
+   * Returns true if this query does not specify any query constraints that could remove results.
+   */
+  public boolean matchesAllDocuments() {
+    return filters.isEmpty() && limit == NO_LIMIT && startAt == null && endAt == null;
```

Comment:
I didn't feel like there was much to gain from special-casing this, but since this is now a generalized API in Query itself, it probably makes sense to address this case.